File: Python For Everyone, Horstmann/Chapter2/P2.19.py
# Write a program that transforms numbers  1 ,  2 ,  3 , …,  12
# into the corresponding month names  January ,  February ,
# March , …,  December . Hint: Make a very long string  "January
# February March ..." , in which you add spaces such that
# each month name has the same length. Then concatenate
# the characters of the month that you want. If you are
# bothered by the trailing spaces, use the  strip method to
# remove them.

userInput = int(input("Enter the month's number: "))

# longest month name is 9 characters, make the space between every month 9 characters,
# including months' names
months = "January  February March    April    May      June     July     August   September" \
         "October  November December "

# start from (userInput - 1 (so we get correct months) * 9) and end 9 characters after that
print("Month's name: %s" % months[(userInput-1) * 9:((userInput-1) * 9) + 9])

File: Big-Java-for-Everyone-Late-Objects-Horstmann/Chapter-2-Fundamental-Data-Types/P2_19.java
/*Write a program that transforms numbers  1 ,  2 ,  3 , …,  12
into the corresponding month names  January ,  February ,
March , …,  December . Hint: Make a very long string  "January
February March ..." , in which you add spaces such that
each month name has the same length. Then use substring to extract the month you want.*/


import java.util.Scanner;


public class P2_19 {
    public static void main(String[] args) {
        Scanner input = new Scanner(System.in);
        System.out.print("Month number: ");
        int month = input.nextInt();
        input.close();
        String month_names = "January  February March    April    May      June     July     August   SeptemberOctober  November December ";
        String month_text = month_names.substring((month - 1) * 9, month * 9);
        System.out.println(month_text);
    }
}
